src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpUpdateProjectToAllowUnsafe.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+using ProjectUtils = Microsoft.VisualStudio.IntegrationTest.Utilities.Common.ProjectUtils;
+
+namespace Roslyn.VisualStudio.IntegrationTests.Other
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpUpdateProjectToAllowUnsafe : AbstractIntegrationTest
+    {
+        private readonly string OriginalProjectFile = $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<Project ToolsVersion=""15.0"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
+  <Import Project=""$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props"" Condition=""Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')"" />
+  <PropertyGroup>
+    <Configuration Condition=""'$(Configuration)' == ''"">Debug</Configuration>
+    <Platform Condition=""'$(Platform)' == ''"">x64</Platform>
+    <ProjectGuid>{{F4233BA4-A4CB-498B-BBC1-65A42206B1BA}}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <RootNamespace>{ProjectName}</RootNamespace>
+    <AssemblyName>{ProjectName}</AssemblyName>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition=""'$(Configuration)|$(Platform)' == 'Debug|x86'"">
+    <OutputPath>bin\x86\Debug\</OutputPath>
+    <PlatformTarget>x86</PlatformTarget>
+  </PropertyGroup>
+  <PropertyGroup Condition=""'$(Configuration)|$(Platform)' == 'Release|x86'"">
+    <OutputPath>bin\x86\Release\</OutputPath>
+    <PlatformTarget>x86</PlatformTarget>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+  <PropertyGroup Condition=""'$(Configuration)|$(Platform)' == 'Debug|x64'"">
+    <OutputPath>bin\x64\Debug\</OutputPath>
+    <PlatformTarget>x64</PlatformTarget>
+    <AllowUnsafeBlocks>false</AllowUnsafeBlocks>
+  </PropertyGroup>
+  <PropertyGroup Condition=""'$(Configuration)|$(Platform)' == 'Release|x64'"">
+    <OutputPath>bin\x64\Release\</OutputPath>
+    <PlatformTarget>x64</PlatformTarget>
+  </PropertyGroup>
+  <ItemGroup>
+  </ItemGroup>
+  <Import Project=""$(MSBuildToolsPath)\Microsoft.CSharp.targets"" />
+</Project>";
+
+        private readonly string ExpectedProjectFile = $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<Project ToolsVersion=""15.0"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
```

Comment:
I'd rather we just read in the file, and do Assert.Contains("<AllowUnsafeBlocks>true</AllowUnsafeBlocks", fileContents). My worry is this will be a maintenance burden, _and_ I also want us testing on current templates.

Reply:
@jasonmalinowski  But the templates only contain 1 platform. This test is specifically added to prevent a regression of this: #23342

Reply:
I don't think this is going to be maintenance burden if we provide our own input csproj.

Reply:
I don't see the burden either. This is just like tests for code changes where we always test the exact code, and we know that changes to formatting might break that.

Reply:
No, then we have the project system team have a test on their side to handle multiple platforms. :smile:

Reply:
No, then we have the project system team test that IVsBuildPropertyStorage work correctly in their repo, since they're the ones who could break it, and the ones who would fix it.

Reply:
> These projects change... what? Every 5 years or so? Even if it was once a year... it just doesn't seem like much of an issue.

The input file is perhaps fine. The output though *definitely* can change due to project system changes. I want us to be immune to changes like that.

Reply:
@jasonmalinowski Yup.  You convinced me.  I'm totally behind your position.

Reply:
@jasonmalinowski So we don't need a regression test?

Reply:
We should:

1. Move to IVsBuildPropertyStorage.
2. Have a test that asserts that the appropriate tag appears in the file after we've done that with a simple Assert.Contains().

Basically, the idea is we're testing up to our boundary, while still giving the project system room to change behavior if they need it.

Reply:
> 1. Move to IVsBuildPropertyStorage.

➡️ Moved to that for CPS projects in c5275a0dc3645eec94a32d37cde131f9b574b40a

> 2. Have a test that asserts that the appropriate tag appears in the file after we've done that with a simple Assert.Contains().

➡️ Fixed by using XML checks (not a simple Assert.Contains though) in 7003d31fd2d7280f3f70cd41cd84f2258ffe288a